### PR TITLE
[wip] Enable CORS on API

### PR DESCRIPTION
**Pull request by zamentur**
## The problem

Try to fix CORS issues on API and moulinette
https://github.com/YunoHost/moulinette/pull/324

## Solution
Use nginx to add some headers

## PR Status

WIP, we might add a settings to define the url that can use the API

## How to test

...

### CODE REVIEW

File: conf/nginx/yunohost_api.conf.inc [hunk 1]
```diff
@@ -6,6 +6,18 @@ location /yunohost/api/ {
...
     proxy_set_header Host $http_host;
 
+    # Rewrite CORS headers due to bug inside moulinette
+    # cause it's very difficult to add them with bottle 
+    # framework 0.12.
+    # Important: to make with credentials working and be
+    # able to transfer cookies from JS. We need to specify
+    # an accurate origin and not just *
+    # https://github.com/YunoHost/moulinette/pull/324
+    more_set_headers "Access-Control-Allow-Origin : https://yunohost.org";
```
[the remark this answers is not in the document]
Comment (zamentur) [reply conf/nginx/yunohost_api.conf.inc:16]: No, as explain in the PR status:"WIP, we might add a settings to define the url that can use the API"

File: conf/nginx/yunohost_api.conf.inc [hunk 2]
```diff
@@ -6,6 +6,18 @@ location /yunohost/api/ {
...
+    more_set_headers "Access-Control-Allow-Headers: x-requested-with";
+    more_set_headers "Allow: GET, POST, PUT, DELETE, OPTIONS";
```
[the remark this answers is not in the document]
Comment (zamentur) [reply conf/nginx/yunohost_api.conf.inc:18]: https://developer.mozilla.org/fr/docs/Web/HTTP/Headers/Allow

[the remark this answers is not in the document]
Comment (zamentur) [reply conf/nginx/yunohost_api.conf.inc:18]: But we can add Access-Control-Request-Method

**Comment by alexAubin** [comment]
Superseded by https://github.com/YunoHost/yunohost/commit/09c5a4cfb91cdede55956bb2e3cf747a0a2c6e18

**Pull request closed without merging**